# DESAFIO 03  -  (Tolueno-Benzeno)

Para o Desafio03, precisamos fazer os compostos entrarem em combustão para posteriormente, no desafio 04, entrar no ciclo de Rankine

In [1]:
#importar bibliotecas
import numpy as np
from uncertainties import ufloat

### **O que precisamos :**

<dt>
    
* **POTENCIAIS TERMODINÂMICOS** (envolvidos na reação de combustão)
    - energia interna
    - entalpia
    - energia livre de Gibbs 
    
<DT>
    
* **ENTROPIA ENVOLVIDA NA REAÇÃO DE COMBUSTÃO** <DT>

* **EQUAÇÃO DE ESTADO A SER USADO NO SISTEMA** <DT>

Baseado no desafio 1, escolhemos utilizar uma concentração de 2 para 1 de tolueno e benzeno respectivamente. Tal concentração foi encontrada através do balançeamento da reação química<br>

$ 0.8 \: C_6 H_5 CH_3 + 0.4 \: C_6H_6 + 20.4 \: O_2 \rightleftharpoons 4.4 \: H_2O + 8 \: CO_2 $

### 1) Cálculo da Entalpia de combustão por meio da entalpia de formação

#### Entalpias de formação (retiradas de https://www.nist.gov/):

In [2]:
ENTALPIA_DE_FORMACAO_BENZENO = ufloat(49.0, 0.5)
ENTALPIA_DE_FORMACAO_TOLUENO = ufloat(12.0, 0.6)
ENTALPIA_DE_FORMACAO_O2 = 0
ENTALPIA_DE_FORMACAO_H2O = ufloat(-285.83, 0.04)
ENTALPIA_DE_FORMACAO_CO2 = ufloat(-393.5, 0.1)

#### Benzeno:

Considerando a reação:<br>
$ 2 \: C_6H_6 + 15 \: O_2 \rightleftharpoons 6 \: H_2O + 12 \: CO_2 $

In [3]:
# Para 2 mol:
entalpia_produtos = 6 * ENTALPIA_DE_FORMACAO_H2O + 12 * ENTALPIA_DE_FORMACAO_CO2
entalpia_reagentes = 2 * ENTALPIA_DE_FORMACAO_BENZENO + 15 * ENTALPIA_DE_FORMACAO_O2
entalpia_combustao_benzeno_2 = entalpia_produtos - entalpia_reagentes
# Para 1 mol:
entalpia_combustao_benzeno = entalpia_combustao_benzeno_2/2

print(f'A entalpia de combustão do benzeno é {entalpia_combustao_benzeno} kJ/mol')

A entalpia de combustão do benzeno é -3267.5+/-0.8 kJ/mol


Nota-se que este valor condiz com a literatura. De acordo com [5]: $ \Delta H_c = -3267.5 \pm 0.42 kJ/mol$

#### Tolueno:

Considerando a reação:<br>
$ C_6 H_5 CH_3 + 9\: O_2 \rightleftharpoons 7\: CO_2 + 4\: H_2O $

In [4]:
# Para 1 mol:
entalpia_produtos = 4 * ENTALPIA_DE_FORMACAO_H2O + 7 * ENTALPIA_DE_FORMACAO_CO2
entalpia_reagentes = 1 * ENTALPIA_DE_FORMACAO_TOLUENO + 9 * ENTALPIA_DE_FORMACAO_O2
entalpia_combustao_tolueno = entalpia_produtos - entalpia_reagentes

print(f'A entalpia de combustão do tolueno é {entalpia_combustao_tolueno} kJ/mol')

A entalpia de combustão do tolueno é -3909.8+/-0.9 kJ/mol


Nota-se que este valor também condiz com a literatura. De acordo com [5]: $ \Delta H_c = -3920. \pm 20 $ kJ/mol

#### Mistura:

A partir da reação de combustão da mistura: <br>
$ 0.8 \: C_6 H_5 CH_3 + 0.4 \: C_6H_6 + 20.4 \: O_2 \rightleftharpoons 4.4 \: H_2O + 8 \: CO_2 $ <br>

In [5]:
# Para 1.2 mol de mistura:
entalpia_produtos = 4.4 * ENTALPIA_DE_FORMACAO_H2O + 8 * ENTALPIA_DE_FORMACAO_CO2
entalpia_reagentes = 0.8 * ENTALPIA_DE_FORMACAO_TOLUENO + 0.4 * ENTALPIA_DE_FORMACAO_BENZENO + 20.4 * ENTALPIA_DE_FORMACAO_O2
entalpia_combustao_mistura_12 = entalpia_produtos - entalpia_reagentes
# Para 1 mol:
entalpia_combustao_mistura = entalpia_combustao_mistura_12/1.2

print(f'A entalpia de combustão da mistura é {entalpia_combustao_mistura} kJ/mol')

A entalpia de combustão da mistura é -3695.7+/-0.8 kJ/mol


#### Portanto, concluímos que, para 1 mol de mistura, podemos fornecer 3696 kJ de energia para o ciclo de Rankhine.

### 2) Cálculo da variação de entropia 

#### Entropias dos compostos (retiradas de https://www.nist.gov/):

In [6]:
ENTROPIA_BENZENO = -24.084
ENTROPIA_TOLUENO = -42.900
ENTROPIA_O2 = 205.01
ENTROPIA_H2O = 6.5773
ENTROPIA_CO2 = 120.43

#### Benzeno:

Considerando a reação:<br>
$ 2 \: C_6H_6 + 15 \: O_2 \rightleftharpoons 6 \: H_2O + 12 \: CO_2 $

In [7]:
# Para 2 mol:
entropia_produtos = (6 * ENTROPIA_H2O) + (12 * ENTROPIA_CO2)
entropia_reagentes = (2 * ENTROPIA_BENZENO) + (15 * ENTROPIA_O2)
entropia_combustao_benzeno_2 = entropia_produtos - entropia_reagentes
# Para 1 mol:
entropia_combustao_benzeno = entropia_combustao_benzeno_2/2

print(f'A variação de entropia da combustão do benzeno é {np.round(entropia_combustao_benzeno,3)} J/mol K')

A variação de entropia da combustão do benzeno é -771.179 J/mol K


#### Tolueno:

Considerando a reação:<br>
$ C_6 H_5 CH_3 + 9\: O_2 \rightleftharpoons 7\: CO_2 + 4\: H_2O $

In [8]:
# Para 1 mol:
entropia_produtos = (4 * ENTROPIA_H2O) + (7 * ENTROPIA_CO2)
entropia_reagentes = (1 * ENTROPIA_TOLUENO) + (9 * ENTROPIA_O2)
entropia_combustao_tolueno = entropia_produtos - entropia_reagentes

print(f'A variação da entropia de combustão do tolueno é {np.round(entropia_combustao_tolueno, 3)} J/mol K')

A variação da entropia de combustão do tolueno é -932.871 J/mol K


#### Mistura:

A partir da reação de combustão da mistura: <br>
$ 0.8 \: C_6 H_5 CH_3 + 0.4 \: C_6H_6 + 20.4 \: O_2 \rightleftharpoons 4.4 \: H_2O + 8 \: CO_2 $ <br>

In [9]:
# Para 1.2 mol de mistura:
entropia_produtos = (4.4 * ENTROPIA_H2O) + (8 * ENTROPIA_CO2)
entropia_reagentes = (0.8 * ENTROPIA_TOLUENO) + (0.4 * ENTROPIA_BENZENO) + (20.4 * ENTROPIA_O2)
entropia_combustao_mistura_12 = entropia_produtos - entropia_reagentes
# Para 1 mol:
entropia_combustao_mistura = entropia_combustao_mistura_12/1.2

print(f'A variação da entropia de combustão da mistura é {np.round(entropia_combustao_mistura, 3)} J/mol K')

A variação da entropia de combustão da mistura é -2621.559 J/mol K


### Energia de Gibbs:

Energia de Gibbs é dada por:<br>
$\Delta G = \Delta H - T \Delta S$

Já calculamos que $\Delta H = -3695.7\pm 0.8$ kJ/mol e que $\Delta S = -2621.559$ J/mol K ou $\Delta S = -2.622$ kJ/mol, e estamos considerando a temperatura de 298 K, logo:

In [10]:
T = 298
gibbs_combustao_banzeno = entalpia_combustao_benzeno - 298 * 1e-3 * entropia_combustao_benzeno 
gibbs_combustao_tolueno = entalpia_combustao_tolueno - 298 * 1e-3 * entropia_combustao_tolueno
gibbs_combustao_mistura = entalpia_combustao_mistura - 298 * 1e-3 * entropia_combustao_mistura

In [11]:
print(f'A energia livre de Gibbs da reação de combustão do benzeno é {gibbs_combustao_banzeno} kJ')
print()
print(f'A energia livre de Gibbs da reação de combustão do tolueno é {gibbs_combustao_tolueno} kJ')
print()
print(f'A energia livre de Gibbs da reação de combustão da mistura é {gibbs_combustao_mistura} kJ')
print()

A energia livre de Gibbs da reação de combustão do benzeno é -3037.7+/-0.8 kJ

A energia livre de Gibbs da reação de combustão do tolueno é -3631.8+/-0.9 kJ

A energia livre de Gibbs da reação de combustão da mistura é -2914.5+/-0.8 kJ



#### Observa-se que a enrgia livre de Gibbis é negativa para todas as reações. Isso indica a espontaneidade da reação, ou seja, ambos os reagentes são inflamáveis.

entropia do benzeno: https://webbook.nist.gov/cgi/fluid.cgi?T=+298&PLow=0.9&PHigh=1.1&PInc=&Digits=5&ID=C71432&Action=Load&Type=IsoTherm&TUnit=K&PUnit=atm&DUnit=mol%2Fl&HUnit=kJ%2Fmol&WUnit=m%2Fs&VisUnit=uPa*s&STUnit=N%2Fm&RefState=DEF

entropia O2:https://webbook.nist.gov/cgi/fluid.cgi?T=+298&PLow=1&PHigh=1&PInc=&Digits=5&ID=C7782447&Action=Load&Type=IsoTherm&TUnit=K&PUnit=atm&DUnit=mol%2Fl&HUnit=kJ%2Fmol&WUnit=m%2Fs&VisUnit=uPa*s&STUnit=N%2Fm&RefState=DEF

entropia CO2:https://webbook.nist.gov/cgi/fluid.cgi?T=298&PLow=1&PHigh=1&PInc=&Digits=5&ID=C124389&Action=Load&Type=IsoTherm&TUnit=K&PUnit=atm&DUnit=mol%2Fl&HUnit=kJ%2Fmol&WUnit=m%2Fs&VisUnit=uPa*s&STUnit=N%2Fm&RefState=DEF

## Referências

* 3) https://en.wikipedia.org/wiki/Benzene_(data_page) 

* 4) benzeno: https://webbook.nist.gov/cgi/inchi?ID=C71432&Mask=2#Thermo-Condensed
* 5) entalpia de combustão benzeno: https://webbook.nist.gov/cgi/cbook.cgi?ID=C71432&Type=HCOMBL#HCOMBL

* 6) água: https://webbook.nist.gov/cgi/cbook.cgi?ID=C7732185&Mask=2
    
* 7) CO2:https://webbook.nist.gov/cgi/cbook.cgi?ID=C124389&Mask=1 

#### Mateiral para referência no Livro McQuarrie (Physical Chemistry: A Molecular Approach)

Como trabalhar com entaplia:
seções - (19-9 a 19-12)

Como calcular entalpia em sistemas termodinâmicos:
seções - (21-1 a 21-5)

Como calcular a enttropia de uma reação quimica a partir de entropias padrão:
seções - (21-9)

Variação de entalpia, entropia e energia livre de Gibbs em função da temperatura:
seções - (21-1 a 22-7)

Potênciais termodinâmicos:
seções - (22)